src/app/components/content-rating-alert/content-rating-alert.component.ts

```diff
@@ -267,4 +256,47 @@ export class ContentRatingAlertComponent implements OnInit {
     });
   }
 
+  generateContentRatingTelemetry(option) {
+    const viewDismissData = {
+      rating: this.ratingCount ? this.ratingCount : this.userRating,
+      comment: this.allComments ? this.allComments : '',
+      message: ''
+    };
+    this.contentService.sendFeedback(option).subscribe((res) => {
+      viewDismissData.message = 'rating.success';
+      this.popOverCtrl.dismiss(viewDismissData);
+      this.commonUtilService.showToast('THANK_FOR_RATING', false, 'green-toast');
+      if (this.navigateBack) {
+        this.location.back();
+      }
+    }, (data) => {
+      viewDismissData.message = 'rating.error';
+      this.popOverCtrl.dismiss(viewDismissData);
+    });
+  }
+
+  generateContentFeedbackTelemetry(option1) {
+    const opts = this.allComments.split(',');
+    opts.forEach(opt => {
+      const option: TelemetryFeedbackRequest = {
+        objId: this.content.identifier,
+        comments: this.allComments,
```

Comment:
comments should be iterated from this.allComments .
Share  all telemetry events with me .